src/linker/Linker.Dataflow/ReflectionMethodBodyScanner.cs

```diff
@@ -1266,15 +1373,31 @@ void MarkFieldsOnTypeHierarchy (ref ReflectionPatternContext reflectionContext,
 			}
 		}
 
-		void MarkNestedTypesOnType (ref ReflectionPatternContext reflectionContext, TypeDefinition type, Func<TypeDefinition, bool> filter)
+		TypeDefinition[] MarkNestedTypesOnType (ref ReflectionPatternContext reflectionContext, TypeDefinition type, Func<TypeDefinition, bool> filter, BindingFlags bindingFlags = BindingFlags.Default)
```

Comment:
Please change the callsite in here https://github.com/mono/linker/pull/1195/files#diff-d634a3c96357eb6bf5b24034efb1e61aR1253-R1257 to use the BIndingFlags just like all the other kinds... (no need for the filter in those cases).